android/kotlin_guide.md

```diff
@@ -0,0 +1,49 @@
+# Kotlin guide
+This guide is designed to help you get started with [Kotlin], the language we write most of our code in, and to introduce you to some intermediate topics.
+- [Getting Started](#getting-started)
+- [Staying Informed](#staying-informed)
+- [Intermediate Topics](#intermediate-topics)
+
+## Getting Started
+There are multiple ways to get started with Kotlin, depending on how you like to learn. If you like:
+- Reading documentation, the official Kotlin docs are good: start at [Basic Syntax] and work your way through the pages in the side bar
+- Learning while coding, try the [Kotlin Koans] online or in the IDE
+- Courses, try the official [Kotlin for Android Developers][course] course (note: none of our team members have tried this yet)
+- Reading books, try [Kotlin in Action]
+
+Once you've gained some basic familiarity with the language, you can gain more experience by:
+- Watching videos and conference talks on YouTube.
+- [Convert Java to Kotlin with the auto-converter][convert]and change the resulting code into idiomatic Kotlin
+- Doing coding exercises on [exercism.io](https://exercism.io/)
+- Working on Mozilla code ;)
+
+You should also take a look at the official style guides:
+- [Jetbrains](https://kotlinlang.org/docs/reference/coding-conventions.html)
+- [Android](https://android.github.io/kotlin-guides/)
```

Comment:
Btw, do we prefer one of these over the other?  I remember that there are a few discrepancies, although none come to mind at the moment.

Reply:
I think we should create a living Mozilla style guide, we could definitely use one of these as the base.

Reply:
I don't think we have a preference for one over the other: @pocmo would you agree?

> I think we should create a living Mozilla style guide,

I agree: it would be good to document how we use things that aren't explicitly documented elsewhere or not enforced by our tools (I wish the tools were stricter!). I filed #49.

Reply:
Regarding the command line option: https://stackoverflow.com/a/45713741